### ISSUE TITLE
Support ReactiveCocoa with Swift 2.0

### ISSUE THREAD

**Issue by esttorhe** [Moya/Moya#195]
Right now `Moya` compiles and works on its `swift-2.0` branch including `RxSwift` but `ReactiveCocoa` is lagging behind due to a non-official `podspec`.

We need to support `ReactiveCocoa` as a dependency working with `Swift 2.0`

**Comment by mhuusko5** [comment]
Curious what's involved in this? Seems like it's the last piece in the puzzle in getting Moya working under Swift 2/Xcode 7..

Great work, btw!

**Comment by esttorhe** [comment]
Sadly `ReactiveCocoa` dropped support for `CocoaPods` some versions ago.

This means that the `podspec` is not maintained by them and there's currently no `Swift 2.0` `podspec`.

I forked `ReactiveCocoa` and added a `podspec` to their 2.0 branch but I'm having some issues when trying to compile the pod.

TL;DR
I need to figure out a correct `podspec` for `Swift 2.0` :smile:

**Comment by ashfurrow** [comment]
I maintain the podspec for ReactiveCocoa; is it out of date?

**Comment by esttorhe** [comment]
I think is up to date but if I try to point the pod to the `Swift 2.0` branch it fails telling me there's no pod at that location or something along those lines (can't remember the exact error right now).

**Comment by ashfurrow** [comment]
Hmm. You should be

**Comment by esttorhe** [comment]
```console
Fetching podspec for `Moya` from `../`
[!] Unknown external source parameters for `ReactiveCocoa`: `{:branch=>"swift2"}`
```

**Comment by esttorhe** [comment]
Let me check because I just changed something and I might be writing the `Podfile` wrong

**Comment by ashfurrow** [comment]
What's the podfile look like?

**Comment by esttorhe** [comment]
Ok; here's the `Podfile`:
```ruby
platform :ios, '8.0'

source 'https://github.com/CocoaPods/Specs.git'

use_frameworks!

pod 'ReactiveCocoa', :git => 'https://github.com/ReactiveCocoa/ReactiveCocoa', :branch => "swift2"
pod 'RxSwift', :git => 'https://github.com/ReactiveX/RxSwift', :branch => 'rxswift-2.0'
pod 'Alamofire', :git => "https://github.com/Alamofire/Alamofire", :branch => "swift-2.0"

pod 'Moya/RxSwift', :path => "../"
pod 'Moya/ReactiveCocoa', :path => "../"
pod 'Moya', :path => "../"

target 'DemoTests' do

  pod 'Quick', :git => 'https://github.com/Quick/Quick', :branch => 'swift-2.0'
  pod 'Nimble', :git => 'https://github.com/Quick/Nimble', :branch => 'swift-2.0'
  pod 'OHHTTPStubs'

end
```

And here's the error:
```console
Pre-downloading: `ReactiveCocoa` from `https://github.com/ReactiveCocoa/ReactiveCocoa`, branch `swift2`
[!] Unable to find a specification for 'ReactiveCocoa'.
```

**Comment by esttorhe** [comment]
Tried pointing to your fork just in case; this is what I'm getting:
```console
Pre-downloading: `ReactiveCocoa` from `https://github.com/ashfurrow/ReactiveCocoa`, branch `swift2`

[!] Error installing ReactiveCocoa
[!] /usr/local/bin/git clone https://github.com/ashfurrow/ReactiveCocoa /var/folders/xk/378q_x1x4b3289cbr_kjyhzr0000gn/T/d20150821-25846-1w2akmk --single-branch --depth 1 --branch swift2

Cloning into '/var/folders/xk/378q_x1x4b3289cbr_kjyhzr0000gn/T/d20150821-25846-1w2akmk'...
warning: Could not find remote branch swift2 to clone.
fatal: Remote branch swift2 not found in upstream origin
```

**Comment by ashfurrow** [comment]
My fork is _way_ out of date. 

Interesting – I would've thought it would use the spec in trunk. You should be able to point to _your_ work and copy&paste this podspec into it: https://github.com/CocoaPods/Specs/blob/master/Specs/ReactiveCocoa/3.0-RC.1/ReactiveCocoa.podspec.json

**Comment by esttorhe** [comment]
I'm using this: https://github.com/esttorhe/ReactiveCocoa/blob/swift2/ReactiveCocoa.podspec

I'll take a look at the one you provided and see what differs from mine

**Comment by ashfurrow** [comment]
I'm spending time this week to update our app to Swift 2 – I'll try to get this working.

**Comment by esttorhe** [comment]
![](http://i.giphy.com/JDnaQ8qn0Myuk.gif)

**Comment by ashfurrow** [comment]
Initial work is complete. I haven't finished yet, so I can't test, but [this](https://github.com/artsy/eidolon/pull/496/files#diff-4a25b996826623c4a3a4910f47f10c30R59) should help anyone currently stuck. I'm using my fork of ReactiveCocoa, and added a podspec to my branch.

**Comment by ashfurrow** [comment]
I'ved pushed a version of ReactiveCocoa 3 with Swift 2 (Xcode 7 GM) support. Its version number is "3.0.0-swift2" and can be used in Podfiles _and_ in podspecs. 

Subsequently, I've updated the Moya podspec to use this new ReactiveCocoa version. There's a problem with the Carthage compatibility, but I'm hoping to have it wrapped up soon, and then I'll submit a PR.

**Comment by ashfurrow** [comment]
OK, now all that's required for ReactiveCocoa extensions is the following in your podfile

```rb
pod 'Moya/ReactiveCocoa'
```

And that's it! :tada:
